utils/frame/try-runtime/cli/src/commands/on_runtime_upgrade.rs

```diff
@@ -29,12 +30,21 @@ pub struct OnRuntimeUpgradeCmd {
 	#[command(subcommand)]
 	pub state: State,
 
-	/// Execute `try_state`, `pre_upgrade` and `post_upgrade` checks as well.
+	/// Select which optional checks to perform. Selects all when no value is given.
 	///
-	/// This will perform more checks, but it will also makes the reported PoV/Weight be
-	/// inaccurate.
-	#[clap(long)]
-	pub checks: bool,
+	/// - `none`: Perform no checks (default when the arg is not present).
+	/// - `all`: Perform all checks (default when the arg is present).
+	/// - `pre-and-post`: Perform pre- and post-upgrade checks.
+	/// - `try-state`: Perform the try-state checks.
+	///
+	/// Performing any checks will potentially invalidate the measured PoV/Weight.
+	#[clap(long,
+		default_value = "None",
+		default_missing_value = "All",
+		num_args = 0..=1,
```

Comment:
We don't need this or? `default_value` should indicate this?

Reply:
could also use some context on what these flags do, but I believe they help us remain backwards compat.

Reply:
It is needed since otherwise it does not work as flag without arg. The difference in the help print:  
`--checks=<CHECKS>` (no `num_args`)
vs
 `--checks[=<CHECKS>]` (with `num_args`)

I added a comment.